spec/honeycomb/integrations/active_support_spec.rb

```diff
@@ -133,7 +133,91 @@
+          config.notification_events = ["honeycomb.nonspecific_event"]
+          # rubocop:disable Metrics/LineLength
+          config.on_notification_event("honeycomb.test_event") do |_name, span, _payload|
+            span.add_field("the_hive", "queen")
+          end
+          # rubocop:enable Metrics/LineLength
+        end
+      end
+      let!(:client) { Honeycomb::Client.new(configuration: configuration) }
+      let(:event_data) { libhoney_client.events.map(&:data) }
+      let(:event_name) { "honeycomb.test_event" }
+
+      before do
+        ActiveSupport::Notifications.instrument event_name, "honeycomb" => 1 do
+        end
+      end
+
+      it_behaves_like "event data", package_fields: false, additional_fields: [
+        "the_hive",
+      ]
+
+      it "uses the default handler for events without a specific handler" do
+        # rubocop:disable Metrics/LineLength
+        ActiveSupport::Notifications.instrument "honeycomb.nonspecific_event", "honeycomb" => 1 do
+        end
+        # rubocop:enable Metrics/LineLength
+
+        this_event = libhoney_client.events.find do |e|
+          e.data["name"] == "honeycomb.nonspecific_event"
+        end
+
+        expect(this_event).not_to be nil
+        expect(this_event.data["honeycomb.nonspecific_event.honeycomb"]).to eq 1
+      end
+
+      context "with a custom default handler" do
+        let(:configuration) do
+          Honeycomb::Configuration.new.tap do |config|
+            config.client = libhoney_client
+            config.notification_events = ["honeycomb.nonspecific_event"]
+            config.on_notification_event do |_name, span, _payload|
+              span.add_field("the_hive", "bees")
+            end
+            # rubocop:disable Metrics/LineLength
+            config.on_notification_event("honeycomb.test_event") do |_name, span, _payload|
+              span.add_field("the_hive", "queen")
+            end
+            # rubocop:enable Metrics/LineLength
+          end
+        end
+
+        before do
+          # rubocop:disable Metrics/LineLength
+          ActiveSupport::Notifications.instrument "honeycomb.nonspecific_event", "honeycomb" => 1 do
+          end
+          # rubocop:enable Metrics/LineLength
+        end
+
+        it_behaves_like(
+          "event data",
+          package_fields: false,
+          additional_fields: ["the_hive"],
+        )
+
+        # rubocop:disable Metrics/LineLength
+        it "uses the new default handler for events without a specific handler" do
+          expect(event_data).to match_array(
+            [
+              hash_including("name" => "honeycomb.nonspecific_event", "the_hive" => "bees"),
+              hash_including("name" => "honeycomb.test_event", "the_hive" => "queen"),
+            ],
+          )
+        end
+        # rubocop:enable Metrics/LineLength
+      end
+    end
+
     describe "pass ActionController::Parameters as hash" do
+      require "action_controller"
```

Comment:
🤨 Seems suspicious that this needs to be `require`d manually. Would it not have failed in CI?

Reply:
Heh. I don't think it'll fail CI because CI will run `BUNDLE_GEMFILE=rails_52.gemfile bundle exec rspec` and I was just running this _specific_ file which didn't include action controller. I am happy to remove it since it was mainly a "local testing just this file" kind of thing.

Reply:
Yea. Someday I might revisit how the multi-dimensional venn diagram of the test suite is wired, but today is not that day.